### PR TITLE
js: Remove usage of DeprecatedString

### PR DESCRIPTION
**Pull request by evansmal**
This removes some usage of `DeprecatedString` throughout the `js` utility. I have also updated `write_to_file()` to use the newer `Core::Stream::File` API.

### CODE REVIEW

File: Userland/Utilities/js.cpp [hunk 1]
```diff
@@ -296,13 +283,13 @@ static ErrorOr<bool> parse_and_run(JS::Interpreter& interpreter, StringView sour
 
         if (!thrown_value.is_object() || !is<JS::Error>(thrown_value.as_object()))
             return {};
-        auto& traceback = static_cast<JS::Error const&>(thrown_value.as_object()).traceback();
+        const auto& traceback = static_cast<JS::Error const&>(thrown_value.as_object()).traceback();
```
Comment (Hendiadyoin1) [inline Userland/Utilities/js.cpp:286]: We use east-const in this project

File: Userland/Utilities/js.cpp [hunk 2]
```diff
@@ -729,7 +729,7 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
                 }
             }
 
-            editor.set_prompt(prompt_for_level(open_indents));
+            editor.set_prompt(MUST(prompt_for_level(open_indents)).to_deprecated_string());
```
Comment (linusg) [inline Userland/Utilities/js.cpp:721]: Don't use MUST here, use
https://github.com/SerenityOS/serenity/blob/1ce34805f81be6b5c544a7f9aa03505d1b46ff26/AK/Error.h#L152

File: Userland/Utilities/js.cpp [hunk 3]
```diff
@@ -505,11 +505,11 @@ JS_DEFINE_NATIVE_FUNCTION(ScriptObject::print)
 static ErrorOr<void> repl(JS::Interpreter& interpreter)
 {
     while (!s_fail_repl) {
-        DeprecatedString piece = read_next_piece();
+        String piece = TRY(read_next_piece());
```
Comment (linusg) [inline Userland/Utilities/js.cpp:495]: `auto`

File: Userland/Utilities/js.cpp [hunk 4]
```diff
@@ -421,8 +408,8 @@ JS_DEFINE_NATIVE_FUNCTION(ReplObject::save_to_file)
 {
     if (!vm.argument_count())
         return JS::Value(false);
-    DeprecatedString save_path = vm.argument(0).to_string_without_side_effects();
-    if (write_to_file(save_path)) {
+    String save_path = TRY(vm.argument(0).to_string(vm));
```
Comment (linusg) [inline Userland/Utilities/js.cpp:411]: `auto`

**Review by linusg — CHANGES_REQUESTED** [review]
Please squash or reword a couple of those commits, there shouldn't me multiple with the exact same message (`js: Remove DeprecatedString usage`)

**Comment by evansmal** [comment]
@linusg Thanks for the review :^)

I've reworked the commits wih (hopefully) some better commit messages. Let me know what you think.

**Pull request merged**